pkg/api/types.go

```diff
@@ -983,6 +984,17 @@ type SourceStepConfiguration struct {
 	ClonerefsPath string `json:"clonerefs_path"`
 }
 
+// IndexGeneratorStepConfiguration describes a step that creates an index database and
+// Dockerfile to build an operator index that uses the generated database based on
+// bundle names provided in OperatorIndex
+type IndexGeneratorStepConfiguration struct {
+	To PipelineImageStreamTagReference `json:"to,omitempty"`
+
+	// OperatorIndex is a list of the names of the bundle images that the
+	// index will contain in its database.
+	OperatorIndex []string `json:"operator_index,omitempty"`
```

Comment:
Why are we asking the user to provide this? Why would someone ever not want all of the bundles they build to be in their index? Why would we expect them to ask us to build a bundle but not an index?